Hw07 patterns

IgorKostoski opened the pull request.

IgorKostoski committed:
Refactor : Address SnoarLint warning in hw05 and hw06

IgorKostoski committed:
feat Implement message processing homework task
'feat: Implement message processing homework tasks

- Added fields field11, field12, and field13 (ObjectForMessage) to the Message class and its builder.
- Created ProcessorSwapFields11And12 to swap the values of field11 and field12.
- Created ProcessorThrowExceptionOnEvenSecond which throws an exception during even seconds, using java.time.Clock for testability.
- Added unit tests for ProcessorThrowExceptionOnEvenSecond.
- Implemented HistoryListener using deep copies to prevent modification of stored messages.
- Updated and enabled HistoryListenerTest to verify deep copy.
- Updated HomeWork main method to demonstrate the usage of the new processors and HistoryListener.

stvort commented on hw07-Patterns/src/main/java/com/example/processor/ProcessorThrowExceptionOnEvenSecond.java:
```diff
@@ -0,0 +1,39 @@
+public class ProcessorThrowExceptionOnEvenSecond implements Processor {
+    private static final Logger logger = LoggerFactory.getLogger(ProcessorThrowExceptionOnEvenSecond.class);
+
+    private final Clock clock;
```
Я бы лучше интерфейсом закрыл, чтобы можно было разные реализации подсовывать и мокать нормально в тестах

IgorKostoski replied:
> Я бы лучше интерфейсом закрыл, чтобы можно было разные реализации подсовывать и мокать нормально в тестах

@stvort  спасибо !

stvort commented on hw05-byteCode/src/main/java/org/example/aop/proxy/Ioc.java:
```diff
@@ -1,5 +1,9 @@
 package org.example.aop.proxy;
 
+import java.lang.annotation.ElementType;
```
Не должно быть лишних вещей в ПР

IgorKostoski committed:
refactor(processor): Improve PorcessorThrowExceptionOnEvenSecond robustness and clarity
-add null check for the inject Clock in the constructor.
-add null check for the input Message in the proces method.
-add debug/warn logging for processing steps and exceptions.
-add/update Jvadoc comments to clarify the class's purpose, methods,and the use of java.time.Clock for dependency injection and testability
